apps/products-feed/src/modules/category-mapping/categories-fetcher.ts

```diff
@@ -0,0 +1,49 @@
+import { Client } from "urql";
+import {
+  CategoryWithMappingFragmentFragment,
+  FetchCategoriesWithMappingDocument,
+} from "../../../generated/graphql";
+
+export class CategoriesFetcher {
+  constructor(private apiClient: Pick<Client, "query">) {}
+
+  private async fetchRecursivePage(
```

Comment:
Idea: extract a utility for fetching paginated resources recursively. 

I spent some time fixing a nasty bug that was caused by the fact that I didn't fetch all the paginated resources, but rather the `first: 100` and slapped a `todo: add pagination` on it. IMO, given the structure, is always the same, we should capitalize on that common standard.

Reply:
Agree, already mentioned this on our channel - I'll add this to the backlog as 1 story-point task ;)